diff -up qtwebkit-opensource-src-5.1.1/Source/WebCore/WebCore.pri.save_memory qtwebkit-opensource-src-5.1.1/Source/WebCore/WebCore.pri
--- qtwebkit-opensource-src-5.1.1/Source/WebCore/WebCore.pri.save_memory	2013-08-28 11:11:19.218230963 -0500
+++ qtwebkit-opensource-src-5.1.1/Source/WebCore/WebCore.pri	2013-08-28 11:12:34.855410992 -0500
@@ -296,6 +296,8 @@ mac {
 }
 unix:!mac:*-g++*:QMAKE_CXXFLAGS += -fdata-sections
 unix:!mac:*-g++*:QMAKE_LFLAGS += -Wl,--gc-sections
+unix:!mac:*-g++*:!equals(QT_ARCH, powerpc):!equals(QT_ARCH, s390):!equals(QT_ARCH, mips):!equals(QT_ARCH, arm):QMAKE_LFLAGS += -Wl,--no-keep-memory
+unix:!mac:*-g++*:!equals(QT_ARCH, powerpc):!equals(QT_ARCH, s390):!equals(QT_ARCH, mips):!equals(QT_ARCH, arm):QMAKE_LFLAGS += -Wl,--reduce-memory-overheads
 linux*-g++*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 contains(DEFINES, ENABLE_OPENCL=1) {
